kubectl-plugin/pkg/cmd/session/session.go

```diff
@@ -143,8 +143,8 @@ func (options *SessionOptions) Validate() error {
 	if err != nil {
 		return fmt.Errorf("Error retrieving raw config: %w", err)
 	}
-	if len(config.CurrentContext) == 0 {
-		return fmt.Errorf("no context is currently set, use %q to select a new one", "kubectl config use-context <context>")
+	if len(config.CurrentContext) == 0 && (options.configFlags.Context == nil || *options.configFlags.Context == "") {
```

Comment:
It seems that this `if` statement is the same in many files. Maybe we can extract the condition to a utility funciton like `hasKubectlContext`. And then replace here with `if !hasKubectlContext(...)`

Reply:
good idea, done!